Modified to work on 0.91.2 and added functionality

CoMPaTech opened the pull request:
Not sure if the heater 'on' detection works, but will try that in the coming days. If you rather put that in haanna that would be better I guess, or now I've put the generic idea here as I still have to tinker with it.

The current code doesn't seem to work under HASS 0.91.2, this PR solves that for me.

Great work, got Anna today and already loving it (in HASS!)

Also addded display of the current preset (home/away/asleep) - are you thinking of including this component in HASS itself? Updating https://community.home-assistant.io/t/advice-how-to-load-sensor-info-from-plugwise-anna-thermostat-web-interface-xml/87801/4 to have the community check on this as well.

Feedback is welcome!

CoMPaTech committed:
Modified to work on 0.91.2 and added hold_mode (i.e. presets), not sure if heater on/off works yet

CoMPaTech commented:
btw: reason for renaming the directory in custom_components is alignment with the changed naming scheme for custom_components

CoMPaTech commented:
![Schermafbeelding 2019-04-24 om 07 49 14](https://user-images.githubusercontent.com/5751088/56635345-8b852680-6665-11e9-9813-bb967260dbff.png)
Detecting heating works too (as included in the PR)

CoMPaTech committed:
Modified haannna accordingly and reworked operation_mode and heater

CoMPaTech commented:
Beware that I 'assumed' `haanna` to be at 0.6.2 or something likewise in both files (preparing for upstreaming to main homeassistant later on). After these commits I'll submit a PR for https://github.com/laetificat/haanna as well!

Added some more changes to have more features available, also heating display definitely works (state on/off = heating status, operation mode is if schedule is active).

Although shown, operation mode can't be switched just yet. Using service climate.set_hold_mode works to switch presets. climate.set_operation_mode is on the todolist.

For those reading along, I only have a Honyewell valve-switch on 'city-heating' infrastructure, so I'm on "on/off" territory.

CoMPaTech commented:
(the reference to renaming the directory is towards https://developers.home-assistant.io/blog/2019/04/12/new-integration-structure.html - still work to be done as well for upstreaming)

laetificat commented:
Thanks for your pull request.

I'll check it out asap, I'll have to update my environment and test it myself since my Anna has not had much use lately.

Regarding including this component to HASS: it would be great to have it in the core but I think this project is still too fragile. I'd like to get this component as close to the silver quality integration as possible. It still need tests and async loading.

CoMPaTech commented:
No problem @laetificat - found it while ordering it last week and extremely happy with all the work you already performed on it and would love to help out with getting it up to par. When you have time let me know what you think of the additional features. If time allows I might still add the set_operation in - if need be we at least have functioning code and can rework things before adding tests and doing the async loading (the latter I really have to look into).

CoMPaTech committed:
Added .92 compatibility as well

laetificat commented on custom_components/anna/climate.py:
```diff
@@ -12,24 +12,62 @@
 import homeassistant.helpers.config_validation as cv
 
-SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE
+SUPPORT_FLAGS = ( SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE | SUPPORT_HOLD_MODE | SUPPORT_AWAY_MODE )
```
Operation mode and away mode have no implementation. Using any of these options returns errors.

I'm not sure what operation mode is supposed to do, the HASS docs say "[Operation mode to set temperature to. This defaults to current_operation mode if not set, or set incorrectly.](https://www.home-assistant.io/components/climate/#service-climateset_temperature)", but isn't this already achieved by using the hold mode?

I'm also not able to set the hold mode at the moment, there is no option to select "home", "vacation", etc.

CoMPaTech replied:
Operation mode I was looking into to switch schedules, but it's a little more difficult (I know how to activate them again, but there is your custom (or more?) schedule and the learning schedule. Not sure though like you state if using operation mode for that is the best. 

Hold mode can be set but the way hass-climate seems to have this done is using service `climate.set_hold_mode`. 

On changing the support flags, good call - I shouldn't have left the partials in - but not sure either on the implementation.

Hold-mode - seems legit for home/vacation etc. - works using the service call
Operation-mode - was looking into (ab)using this for schedule on/of
Away-mode - the device supports it, but I recon me only having it put up it still needs to 'learn', the way the docs of plugwise state it they use geofencing on their app for this? (Not sure if I particularly like it)

CoMPaTech replied:
Commented these out until we have a decision on how to go from here (code is still in, but not advertising them)

CoMPaTech replied:
One way could be to use 'presets' as 'operation_mode' so users can adjust it from the UI directly and either not bother about the schedules or move those as 'hold_mode'. The way I read the code-comments, code-issues and some on the forum is that hold_mode is about retaining temperature at a set (i.e. hold) point, therefor our presets of Anna would align on hold_mode, right? Question would (or can) be, how (or if) we want to bother with being on schedule or not.

laetificat commented on custom_components/anna/manifest.json:
```diff
@@ -0,0 +1,8 @@
+  "domain": "anna",
+  "name": "Plugwise_Anna",
+  "documentation": "https://github.com/laetificat/anna-ha",
+  "dependencies": ["climate"],
```
Having this set to "climate" will cause my HASS instance to not start, currently running version 0.92.0, removing this will fix it.
```suggestion
  "dependencies": [],
```

CoMPaTech replied:
Didn't push this (apologies) - for some reason when upgrading to 0.92.1 yesterday it did work, then break for me because of this.

laetificat commented on custom_components/anna/__init__.py:
```diff
@@ -0,0 +1 @@
+REQUIREMENTS = ['haanna==0.6.2']
```
Is this still required? I'm asking because the manifest also has a requirements line. I would suggest adding a description in here instead so that the file is not empty.
```suggestion
"""
Plugwise Anna component for HomeAssistant
"""
```

CoMPaTech replied:
I'll retry this part, I added the `__init__.py` yesterday with that statement (even with the manifest in).

CoMPaTech replied:
Seems to work, commit following

laetificat committed:
Update custom_components/anna/manifest.json

Remove breaking dependency

Co-Authored-By: CoMPaTech <CoMPaTech@users.noreply.github.com>

CoMPaTech committed:
Added PlatformNotReady (see haaanna commit) and resolved comments

CoMPaTech commented:
Code pushed, `operation_mode` vs `hold_mode` still open although the first is removed as being actively advertised from the code (together with `away_mode`). Added to the documentation how to activate `hold_mode` (i.e. service-call to `climate.set_hold_mode`) and added checks on validity of these modes (along with validating temperature).

CoMPaTech committed:
Info logging not allowed as per styleguide

CoMPaTech commented:
Tiny ping of remembrance - as per multiple users interested from https://community.home-assistant.io/t/advice-how-to-load-sensor-info-from-plugwise-anna-thermostat-web-interface-xml/87801/22

wburgers commented:
Another ping 😇, would love to see this working in home assistant!

CoMPaTech commented:
@laetificat any option for you to make some time available? Or otherwise allow access to the repo, not really looking forward to forking as indicated by the community, especially on things already committed through PyPi
